src/test/java/core/basesyntax/service/TransitionProcessorImplTest.java

```diff
@@ -0,0 +1,78 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Operation;
+import core.basesyntax.model.Transaction;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TransitionProcessorImplTest {
+    private static TransactionProcessor transactionProcessor;
+    private final Fruit banana = new Fruit("banana");
+    private final Fruit apple = new Fruit("apple");
+
+    @BeforeClass
+    public static void beforeClass() {
+        transactionProcessor = new TransactionProcessorImpl(new OperationStrategyImpl());
+        Storage.fruits.clear();
+    }
+
+    @Test
+    public void process_Ok() {
+        List<Transaction> transactions = getTransactionList();
+        Map<Fruit, Integer> expected = new HashMap<>();
+        expected.put(banana, 110);
+        expected.put(apple, 220);
+        transactionProcessor.process(transactions);
+        Map<Fruit, Integer> actual = Storage.fruits;
+        assertEquals("Result of TransactionProcessorImpl.process doesn't equal to expected",
+                expected, actual);
+    }
+
+    @Test(expected = NullPointerException.class)
```

Comment:
same, we should not expect NPE, lets add some null check for tested method (fix in all such places)